### ISSUE TITLE
Type[type] is not indexable when in Union

### ISSUE THREAD

**Issue by jolaf** [python/mypy#13074]
The following code:
```python3
t = type[str] | type[bytes]
```
produces the following output:
```shell
$ mypy test.py
test.py:1:5: error: Value of type "Type[type]" is not indexable  [index]
    t = type[str] | type[bytes]
        ^
Found 1 error in 1 file (checked 1 source file)
```
This looks like a false positive to me.

```shell
$ mypy --version
mypy 0.961 (compiled: yes)

$ python3 --version
Python 3.10.4

$ lsb_release -d
Description:	Ubuntu 22.04 LTS
```

**Comment by AlexWaygood** [comment]
Duplicate of #12392
